File: dynabuffers-python/tests/usecase/Schema26Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema26(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema26.dbs"))

    def test_defaults(self):
        obj = {
            "text": "123",
            "oneInt": 1,
            "twoFloat": 2.0,
            "threeShort": 3,
            "fourLong": 4,
            "trueBoolean": True,
            "falseBoolean": False
        }
        result = self.engine.deserialize(self.engine.serialize({}))
        self.assertEqual(result, obj)


if __name__ == "__main__":
    unittest.main()

File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema26Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test


class Schema26Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine;

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema26Test::class.java.getResourceAsStream("/schema26.dbs"))
    }

    @Test
    fun `test defaults`() {
        val reference = mapOf<String, Any>(
            "text" to "123",
            "oneInt" to 1,
            "twoFloat" to 2.0f,
            "threeShort" to 3.toShort(),
            "fourLong" to 4L,
            "trueBoolean" to true,
            "falseBoolean" to false
        )
        val serialized = engine.serialize(reference)
        val deserialized = engine.deserialize(serialized)
        Assertions.assertThat(deserialized).isEqualTo(reference)
    }
}
